WireAPI/Tests/WireAPITests/APIs/UserPropertiesAPI/UserPropertiesAPITests.swift

```diff
@@ -0,0 +1,142 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import SnapshotTesting
+import XCTest
+
+@testable import WireAPI
+
+final class UserPropertiesAPITests: XCTestCase {
+
+    private var apiSnapshotHelper: APISnapshotHelper<UserPropertiesAPI>!
+
+    // MARK: - Setup
+
+    override func setUp() {
+        super.setUp()
+        apiSnapshotHelper = APISnapshotHelper { httpClient, apiVersion in
+            let builder = UserPropertiesBuilder(httpClient: httpClient)
+            return builder.makeAPI(for: apiVersion)
+        }
+    }
+
+    override func tearDown() {
+        apiSnapshotHelper = nil
+        super.tearDown()
+    }
+
+    // MARK: - Request generation
+
+    func testGetUserPropertyRequest() async throws {
+        try await apiSnapshotHelper.verifyRequestForAllAPIVersions { sut in
+            _ = try await sut.getProperty(forKey: .wireTypingIndicatorMode)
+        }
+    }
+
+    // MARK: - Response handling
+
+    // MARK: - V0
```

Comment:
How hard would it be to make these tests cover all versions via parametized testing like you did in your other PR?

Reply:
Yep will do thanks, I hadn't merged develop into my branch before